pkg/container-collection/options.go

```diff
@@ -910,3 +911,32 @@ func WithTracerCollection(tc TracerCollection) ContainerCollectionOption {
 		return nil
 	}
 }
+
+// WithProcEnrichment enables an enricher to add process metadata
+func WithProcEnrichment() ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		cc.containerEnrichers = append(cc.containerEnrichers, func(container *Container) bool {
+			pid := int(container.Pid)
+			if pid == 0 {
+				log.Errorf("proc enricher: failed to enrich container %s with pid zero", container.Runtime.ContainerID)
+				return true
+			}
+
+			p, err := process.NewProcess(int32(pid))
```

Comment:
This package uses unix signals to check if the process exists:
https://github.com/shirou/gopsutil/blob/master/process/process_posix.go#L121

And unix signals only work if the process is in the same pid namespace as ig.

This also uses `/proc`. If ig is running in a container, we would need to define the `$HOST_PROC` environment variable to make it use `/host/proc`:
https://github.com/shirou/gopsutil/blob/ac15d685454b02de6fbfddd95fecaf3e50fd45f7/internal/common/common.go#L392

I wonder if it's better to avoid another dependency and just do a `stat` syscall: 
```
stat --printf=%y /proc/$$
```

Reply:
Makes sense. It was too late when I fixed that yesterday, I'll publish a new version using the stat syscall shortly.